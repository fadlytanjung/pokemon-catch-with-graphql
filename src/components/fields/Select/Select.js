import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Select(props) {
  const { className, input, inputProps, meta, options } = props;
  const { dirty, error, touched } = meta;
  const classes = [
    styles.root,
    !!input.value || styles.empty,
    !!error && (dirty || touched) && styles.error,
    className,
  ].filter(Boolean).join(' ');

  return (
    <>
      <label className={styles.label}>{inputProps.label}</label>
      <select className={classes} {...input} {...inputProps}>
        <option disabled value="">{inputProps.placeholder}</option>
        {options.map((i, idx) => (
          <option key={idx} value={i.value}>{i.text}</option>
        ))}
      </select>
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
    </>
  );
}

Select.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  meta: {},
  options: [],
};

Select.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  meta: PropTypes.object,
  options: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    value: PropTypes.string,
  })),
};
