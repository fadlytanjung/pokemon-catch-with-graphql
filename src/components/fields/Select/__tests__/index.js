import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Select from '../Select';

describe('src/components/fields/Select', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Select options={[{ text: '', value: '' }]} />);
    expect(tree).toMatchSnapshot();
  });

  test('error message', () => {
    const props = {
      ...Select.defaultProps,
      meta: {
        error: 'error',
        touched: true,
      },
    };
    const result = Select(props);
    expect(result.props.children[2].props.children).toBe('error');
  });
});
