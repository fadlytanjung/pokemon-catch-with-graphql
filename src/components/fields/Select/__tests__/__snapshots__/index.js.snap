// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Select render 1`] = `
<React.Fragment>
  <label />
  <select
    className=""
  >
    <option
      disabled={true}
      value=""
    />
    <option
      value=""
    >
      
    </option>
  </select>
</React.Fragment>
`;
